### PR TITLE
Refactoring v0.3

### LINKED ISSUE #7: Refactoring for 0.3
- Remove the Interface suffix
- Improve Collection with "withAdded" method
- Refactor with / without APIs

### PR DESCRIPTION
**Pull request by damianopetrungaro**
Fix #7 but should be controlled a little bit more.
@bigfoot90 see that you was interested to the repo, you can give me your opinion too.

### CODE REVIEW

File: examples/slim/src/Users/Application/Repository/DBALUserRepository.php [hunk 1]
```diff
@@ -30,11 +30,12 @@
...
      * @param string $userTable
      * @param Connection $connection
-     * @param UserMapperInterface $mapper
+     * @param UserMapper $mapper
      */
-    public function __construct(Connection $connection, UserMapperInterface $mapper, string $userTable)
+    public function __construct(Connection $connection, UserMapper $mapper, string $userTable)
     {
         $this->connection = $connection;
         $this->mapper = $mapper;
```
Comment (bigfoot90) [inline examples/slim/src/Users/Application/Repository/DBALUserRepository.php:41]: Property `$this->mapper` is type of `Mapper`,
but the parameter `$mapper` in the constructor is type of `UserMapper`

File: src/Common/Enum/Enum.php [hunk 2]
```diff
@@ -1,69 +1,51 @@
...
-declare(strict_types = 1);
+declare(strict_types=1);
 
 namespace Damianopetrungaro\CleanArchitecture\Common\Enum;
 
-use ReflectionClass;
+use InvalidArgumentException;
 
 /**
  * The class that will extents Enum should only contains constant.
```
Comment (bigfoot90) [inline src/Common/Enum/Enum.php]: use `that will extend` or `that extends`

File: src/UseCase/Response/Response.php [hunk 3]
```diff
@@ -1,150 +1,104 @@
...
-     * {@inheritDoc}
+     * Return the data list content.
+     *
+     * @return array
      */
-    public function getData() : array
-    {
-        return $this->data->all();
-    }
+    public function getData(): array;
```
Comment (bigfoot90) [inline src/UseCase/Response/Response.php:62]: What king of data this returns? the body? the headers?

**Review by bigfoot90 — COMMENTED** [review]
I like it, seems good form me.

What do you think about adding constants for Request's methods and Response's status codes like symfony does?

**Comment by damianopetrungaro** [comment]
The Request and Response that you see is not HTTP Messages.

They are only domain objects by design, if I put http status code those object won't be useful anymore as any PSR7 implementation.

Give me some minutes and I'll see the other changes you suggested.

File: src/UseCase/Response/Response.php (hunk 3 above)
Comment (damianopetrungaro) [reply src/UseCase/Response/Response.php:62]: Return the data (no-http here as mentioned in the reply) that your use case expects to return.

**Comment by bigfoot90** [comment]
Can we find a shorter namespace than `Damianopetrungaro\CleanArchitecture\...` ?

File: src/Common/Enum/Enum.php (hunk 2 above)
Comment (damianopetrungaro) [reply src/Common/Enum/Enum.php]: I'll change this :)

**Comment by bigfoot90** [comment]
Do you have any idea how to integrate this with Symfony framework? so I can try/test this on my projects?

**Comment by damianopetrungaro** [comment]
The namespace is long but imho is not a problem, any IDE or smart text editor import automatically for you.

**Comment by damianopetrungaro** [comment]
You can see the "example" and see how it should be used.
If you have any doubt for use it open a specific issue, so if i understand what is not clear a little more documentation may help :D

File: examples/slim/src/Users/Application/Repository/DBALUserRepository.php (hunk 1 above)
Comment (damianopetrungaro) [reply examples/slim/src/Users/Application/Repository/DBALUserRepository.php:41]: Fixed also the example directory :)

**Pull request merged**